modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java

```diff
@@ -200,7 +202,11 @@ protected DDMForm getDDMForm(DDMFormInstance ddmFormInstance)
 
 		DDMStructure ddmStructure = ddmFormInstance.getStructure();
 
-		return ddmStructure.getDDMForm();
+		DDMStructureVersion ddmStructureVersion =
+			_ddmStructureVersionLocalService.getLatestPublishedStructureVersion(
+				ddmStructure.getStructureId());
+
+		return ddmStructureVersion.getDDMForm();
```

Comment:
Hey @icgsbr 
I was doing research on Forms, and I found this code:
```
DDMFormInstanceVersion latestDDMFormInstanceVersion =
	_ddmFormInstanceVersionLocalService.
	        getLatestFormInstanceVersion(
		        ddmFormInstance.getFormInstanceId(),WorkflowConstants.STATUS_APPROVED);

DDMStructureVersion ddmStructureVersion =
	latestDDMFormInstanceVersion.getStructureVersion();

ddmForm = ddmStructureVersion.getDDMForm();
```
I believe this code does what you are looking for (the latest published structure version). 
Could you try it? If it works, you won't have to add a new method.